Mathlib/Data/List/Basic.lean

```diff
@@ -2983,6 +2983,33 @@ theorem filter_false (l : List α) :
     filter (fun _ => false) l = [] := by induction l <;> simp [*, filter]
 #align list.filter_false List.filter_false
 
+theorem filter_eval_true (l : List α)
+    (h : ∀ a ∈ l, p a = true) : List.filter p l = l := by
+  match l with
+  |[] => simp
+  |x :: xs =>
+    simp [filter]
+    rw [h]
+    simp
+    apply filter_eval_true
+    intro b hb
+    simp at h
+    tauto
+    simp
+
+theorem filter_eval_false (l : List α)
+    (h : ∀ a ∈ l, p a = false) : List.filter p l = []:= by
```

Comment:
This is subsumed by [`List.filter_eq_nil`](https://leanprover-community.github.io/mathlib4_docs/Init/Data/List/Lemmas.html#List.filter_eq_nil)